dtos/deviceprofile.go

```diff
@@ -27,7 +28,8 @@ type DeviceProfile struct {
 func (dp *DeviceProfile) Validate() error {
 	err := common.Validate(dp)
 	if err != nil {
-		return errors.NewCommonEdgeX(errors.KindContractInvalid, "invalid DeviceProfile.", err)
+		err = errors.New(strings.ReplaceAll(err.Error(), ".DeviceProfileBasicInfo", ""))
```

Comment:
add a comment before this line
` // The DeviceProfileBasicInfo is the internal struct in Golang programming, not in the Profile model, so it should be hidden from the error messages.`